### PR TITLE
Testing cabal init for buildable output

### LINKED ISSUE #7689: `cabal init` tests should check for buildable output
As of now the tests pertaining `cabal init` only check if the command generates the rightful stanzas and .cabal file, however none of them check if the generated file can actually build a package proper. Such tests would go a long way to prevent bugs like #7680.

### PR DESCRIPTION
**Pull request by ptkato**
Closes #7689.

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

### CODE REVIEW

File: cabal-testsuite/PackageTests/Init/init.test.hs [hunk 1]
```diff
@@ -3,6 +3,11 @@ import Test.Cabal.Prelude
 main = cabalTest $
   withSourceCopyDir "app" $ do
     cwd <- fmap testSourceCopyDir getTestEnv
-    cabal "init" ["-n", "--exe", "--application-dir=app", "--main-is=Main.hs", "--", cwd]
+
+    buildOut <- withDirectory cwd $ do
```
Comment (ptkato) [inline cabal-testsuite/PackageTests/Init/init.test.hs:7]: I'm not entirely sure on how `withSourceCopyDir` behaves, so the use of `withDirectory` might be reduntant.

Comment (Mikolaj) [reply cabal-testsuite/PackageTests/Init/init.test.hs:7]: Me neither. As long as it works. :)

**Review by Mikolaj — APPROVED** [review]
Thank you!

**Comment by Mikolaj** [comment]
Some random breakage in CI, so I restarted it.

**Comment by Mikolaj** [comment]
Huh, `Unable to locate package python3.8`. Perhaps not so random. If it doesn't pass in a day, let's investigate and fix.

**Comment by Mikolaj** [comment]
Rebasing to enable the fix to CI.

**Comment by Mikolaj** [comment]
@mergifyIO rebase

**Comment by Mikolaj** [comment]
Rebasing after CI really fixed.

**Comment by Mikolaj** [comment]
@Mergifyio rebase

**Pull request merged**